jina/serve/runtimes/head/__init__.py

```diff
@@ -342,31 +371,43 @@ async def _handle_data_request(
         response_request = worker_results[0]
         uses_after_metadata = None
         if self.uses_after_address:
-            (
-                response_request,
-                uses_after_metadata,
-            ) = await self.connection_pool.send_requests_once(
+            uses_after_result = await self.connection_pool.send_requests_once(
                 worker_results,
                 deployment='uses_after',
                 timeout=self.timeout_send,
                 retries=self._retries,
             )
+            if isinstance(uses_after_result, (AioRpcError, InternalNetworkError)):
+                raise uses_after_result
+            else:
+                (response_request, uses_after_metadata) = uses_after_result
         elif len(worker_results) > 1 and self._reduce:
-            WorkerRequestHandler.reduce_requests(worker_results)
+            response_request = WorkerRequestHandler.reduce_requests(worker_results)
```

Comment:
tbh since the operation modifies in place, I think it's more explicit if we don't rely on the return type and keep
WorkerRequestHandler.reduce_requests(worker_results)
I'm also fine with removing the return for the function

Reply:
I won't be able to fix this now. It's taking too much time to go through all the mutations that are being used. The logic in Head or the `topology_graph.py` must also be updated matching the in place edit.